src/ui-test/camelk_java_debug_test.ts

```diff
@@ -0,0 +1,245 @@
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+import { assert } from 'chai';
+import path = require('path');
+import * as pjson from '../../package.json';
+import {
+	ActivityBar,
+	CustomTreeSection,
+	DebugToolbar,
+	EditorView,
+	InputBox,
+	SideBarView,
+	TextEditor,
+	ViewItem,
+	VSBrowser,
+	Workbench
+} from 'vscode-extension-tester';
+import { Marketplace } from 'vscode-uitests-tooling';
+import { prepareEmptyTestFolder } from './utils/resourcesUtils';
+import { extensionIsActivated, sectionHasItem } from './utils/waitConditions';
+import * as uiTestConstants from './utils/uiTestConstants';
+
+const TEST_FOLDER = '../../../testFolder';
+const WORKSPACE_FOLDER = path.join(__dirname, TEST_FOLDER);
+
+const START_DEBUG_LABEL = uiTestConstants.startDebug;
+const REMOVE_INTEGRATION_LABEL = uiTestConstants.integrationRemove;
+
+describe('Test Debug on Camel K Integrations from Side Bar', function () {
+
+	before(async function () {
+		this.timeout(90000);
+		this.retries(3);
+		await VSBrowser.instance.driver.sleep(20000);
+		await VSBrowser.instance.waitForWorkbench;
+		await prepareTempWorkspaceForTests(WORKSPACE_FOLDER);
+		await VSBrowser.instance.driver.wait(camelKToolingIsEnabled);
+	});
+
+	describe('Java Debug', function () {
+		
+		const INTEGRATION_LABEL = 'java-debug-test';
+		const INTEGRATION_FILE = 'JavaDebugTest';
+
+		before(async function (){
+			this.timeout(20000);
+			await moveToExplorerActivity();
+			await createIntegration(INTEGRATION_FILE);
+			await startIntegration(INTEGRATION_LABEL);
+		})
+
+		it('Check Java Debug available', async function () {
+			this.timeout(20000);
+
+			const item = await getIntegration(INTEGRATION_LABEL);
+			const menu = await item.openContextMenu();
+
+			assert.isTrue(await menu.hasItem(START_DEBUG_LABEL));
+		});
+
+		it.skip('Check Java Debug stops at breakpoint', async function() {
+			//Currently failing because "No Java Debugger configured" error window
+			//It won't work even if I install the extension using ExTester or manually on the VSCode Instance
+			this.timeout(30000);
+			await addBreakpointToCurrentFile();
+			await startDebug(INTEGRATION_LABEL);
+			assert.isTrue(isThereSomethingInVariablesInDebugger());
```

Comment:
this will always be false because we are looking if a Promise is true.
```suggestion
			assert.isTrue(await isThereSomethingInVariablesInDebugger());
```